src/utils/ProviderUtils.ts

```diff
@@ -185,6 +203,37 @@ export async function getProvider(
   return provider;
 }
 
+/**
+ * @notice Returns a Viem custom transport that can be used to create a Viem client from our customized Ethers
+ * provider. This allows us to send requests through our RetryProvider that need to be handled by Viem SDK's.
+ */
+export function createViemCustomTransportFromEthersProvider(providerChainId: number): viem.CustomTransport {
+  return viem.custom(
+    {
+      async request({ method, params }) {
+        const provider = getCachedProvider(providerChainId, true);
+        try {
+          return await provider.send(method, params);
+        } catch (error: any) {
```

Comment:
Does this work with `unknown` instead of `any` ?

```suggestion
        } catch (error: unknown) {
```